app/service/family.py

```diff
@@ -58,7 +58,7 @@ def all() -> list[FamilyReadDTO]:
 
 
 @validate_call(config=ConfigDict(arbitrary_types_allowed=True))
-def search(query_params: dict[str, str]) -> list[FamilyReadDTO]:
+def search(query_params: dict[str, Union[str, int]]) -> list[FamilyReadDTO]:
```

Comment:
So my preference is that the responsibility for processing/parsing the query params lies within the router. This then allows the service and the repo to be more rigorously typed, and removes the need for them to understand what and how params are sent over HTTP ... what do you think?